### DISCUSSION TITLE
[tip or bug?] default slots in button align horizontal

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2196] [Q&A] [answered]
### Question

As we know
```python
from nicegui import ui

ui.icon("help")
ui.markdown('hello').props('align="left"')

ui.run()
```
places elements vertical

however, code
```python
from nicegui import ui

with ( ui.button()
    .props('autofocus outlined rounded item-aligned input-class="ml-3" align="left" no-caps ')
    .classes('w-full')
):
    ui.icon("help")
    ui.markdown('hello').props('align="left"')

ui.run()
```
places elements in button along horizontal direction, else if we write
```python
with ( ui.button()
    .props('autofocus outlined rounded item-aligned input-class="ml-3" align="left" no-caps ')
    .classes('w-full')
):
    with ui.column():
        ui.icon("help")
        ui.markdown('hello').props('align="left"')
```

**Comment by falkoschindler** [comment, marked as answer]
So the question is, why elements nested in a `ui.button` are aligned horizontally. Well, why not? Even though we decided elements on a page should stack vertically using a flex box layout, other elements can behave differently. In PR https://github.com/zauberzeug/nicegui/pull/1796 we made sure to use flex boxes for all common elements that are expected to contain lots of content. But a button is usually defined using initializer arguments rather than child elements.
